libcloud/dns/drivers/rackspace.py

```diff
@@ -667,9 +667,10 @@ def _rackspace_result_has_more(response, result_length, limit):
 
 
 def _check_ptr_extra_fields(device_or_record):
-    if not (getattr(device_or_record, 'extra') and
```

Comment:
On a related note, I find this if a bit hard to read and digest.

I would find something like this a bit more easy to understand:

```python
extra = getattr(device_or_record, 'extra', {})
uri = extra.get('uri', None)
service_name = extra.get('service_name', None)

if not uri and not service_name:
    # throw..
```